### ISSUE TITLE
E0057 needs to be updated to new format

### ISSUE THREAD

**Issue by jonathandturner** [rust-lang/rust#35214]
From: src/test/compile-fail/E0057.rs

Error E0057 needs a span_label, updating it from:

```
error[E0057]: this function takes 1 parameter but 2 parameters were supplied
  --> src/test/compile-fail/E0057.rs:15:13
   |
15 |     let c = f(2, 3); //~ ERROR E0057
   |             ^^^^^^^
   |
   = note: the following parameter type was expected: (_,)
```

To:

```
error[E0057]: this function takes 1 parameter but 2 parameters were supplied
  --> src/test/compile-fail/E0057.rs:15:13
   |
15 |     let c = f(2, 3); //~ ERROR E0057
   |             ^^^^^^^ expected 1 parameter
   |
   = note: the following parameter type was expected: (_,)
```

Bonus 1: The note seems excessive.  Check other E0057 errors.  If it is not needed, remove the note.

Bonus 2: The span for the error could likely focus on the parameters rather than the whole call.

**Comment by paulfanelli** [comment]
I'm taking this one.

**Comment by paulfanelli** [comment]
@jonathandturner  I think I've got most of this fix/working, but I'm having a problem fixing one of the unit test cases - issue-3044.rs.  The other related test cases (issue-18819, issue-4935, method-call-err-msg, not-enough-arguments, overloaded-calls-bad and variadic-ffi-3) I was able to fix by following your blog guide.  For some reason, issue-3044 is 'different' and I don't understand why.  Not sure how to proceed...

**Comment by jonathandturner** [comment]
@paulfanelli - can you show the errors you get when you try to get issue-3044 to pass?

**Comment by paulfanelli** [comment]
@jonathandturner  I think I fixed the problem.  I'm now trying to get my code checked in and pushed to by fork.  Having some ssh key/cert issues.  I think once I do that, I can do a PR?

**Comment by paulfanelli** [comment]
I just submitted a PR, but not sure if I did it correctly.

**Comment by acrrd** [comment]
The note of #35290 and this one are printed by the same function, can I modify it to print the expected types directly in the label?

I also did the bonus2 part for both E0060 and E0061.

**Comment by bstrie** [comment]
Freeing this one up as part of the final push for #35233.

**Comment by bstrie** [comment]
Actually, according to the comments in https://github.com/rust-lang/rust/pull/35395 maybe this is already closed? @paulfanelli @jonathandturner , can you confirm?

**Comment by KiChjang** [comment]
@bstrie I think #35395 should be closed, and progress is under way in #36222.
